loam/data.lisp

```diff
@@ -81,15 +81,17 @@
 (defstruct (fun (:constructor fun (args body closed-env)))
   (args nil :type list)
   (body nil :type t)
-  (closed-env nil :type list))
+  (closed-env nil :type maybe-env))
 
 (defun tag (thing)
   (etypecase thing
     (null :sym) ; nil is also a sym.
     (boolean :sym) ; nil and t are both sym.
     (cons :cons)
     (keyword :key)
-    (symbol (if (lurk-builtin-p thing) :builtin :sym))
+    (symbol (if (eql 'nil-env thing)
+		:env
+		(if (lurk-builtin-p thing) :builtin :sym)))
```

Comment:
This is probably okay for now, but really I think there should be a struct called `empy-env` (or else the existing `env` struct should be markable as empty). Using a symbol like this is not really correct, since it means that symbol cannot be used as a symbol.